src/main/java/mods/railcraft/common/plugins/forge/WorldPlugin.java

```diff
@@ -140,12 +140,11 @@ public static boolean playerRemoveBlock(World world, BlockPos pos, @Nullable Ent
         if (MinecraftForge.EVENT_BUS.post(new BlockEvent.BreakEvent(world, pos, state, player)))
             return false;
 
-        if (state.getBlock().removedByPlayer(state, world, pos, player, dropBlock))
+        if (!state.getBlock().removedByPlayer(state, world, pos, player, dropBlock))
             return false;
 
         if (dropBlock) {
-            List<ItemStack> drops = state.getBlock().getDrops(world, pos, state, 0);
-            InvTools.dropItems(drops, world, pos);
+            state.getBlock().harvestBlock(world, player, pos, state, getBlockTile(world, pos), player.getHeldItem(player.getActiveHand()));
```

Comment:
From the javadoc of `Block#removedByPlayer`:
```
    * @param willHarvest True if Block.harvestBlock will be called after this, if the return in true.
    *        Can be useful to delay the destruction of tile entities till after harvestBlock
```
The method I call above allows blocks to guarantee that the tile would not get destroyed until `#harvestBlock` is called.